src/uct/ib/base/ib_iface.c

```diff
@@ -923,8 +923,26 @@ static ucs_status_t uct_ib_iface_init_lmc(uct_ib_iface_t *iface,
     return UCS_OK;
 }
 
+static void
+uct_ib_iface_adjust_qp_attr(uct_ib_iface_t *iface, uct_ib_qp_attr_t *qp_attr)
+{
+    uct_ib_device_t *dev    = uct_ib_iface_device(iface);
+    struct ibv_qp_cap *caps = &qp_attr->ibv.cap;
+
+    uct_ib_iface_check_cap_limit(iface, IBV_DEV_ATTR(dev, max_qp_wr),
+                                 "max_send_wr", &caps->max_send_wr);
+    uct_ib_iface_check_cap_limit(iface, IBV_DEV_ATTR(dev, max_qp_wr),
+                                 "max_recv_wr", &caps->max_recv_wr);
+    uct_ib_iface_check_cap_limit(iface, IBV_DEV_ATTR(dev, max_sge),
+                                 "max_send_sge", &caps->max_send_sge);
+    uct_ib_iface_check_cap_limit(iface, IBV_DEV_ATTR(dev, max_sge),
+                                 "max_recv_sge", &caps->max_recv_sge);
```

Comment:
seems like not all users of create_qp consider that the QP may be created with smaller values than passed
for example:
- https://github.com/openucx/ucx/blob/2b468968d6920f7f8c2f5076b260b690ebd70172/src/uct/ib/rc/verbs/rc_verbs_ep.c#L609
- https://github.com/openucx/ucx/blob/e4d67c0431791f58bb1c16fe883d791019769064/src/uct/ib/ud/base/ud_iface.c#L467
- https://github.com/openucx/ucx/blob/1fd6922226460fa6613f1e4fb2b8eeda67e1731a/src/uct/ib/ud/verbs/ud_verbs.c#L477